File: app/src/main/java/com/icerrate/popularmovies/utils/FormatUtils.java
package com.icerrate.popularmovies.utils;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * @author Ivan Cerrate.
 */

public class FormatUtils {

    public static final String FORMAT_yyyy_MM_dd = "yyyy-MM-dd";

    public static final String FORMAT_MMM_dd_yyyy = "MMM', 'dd' 'yyyy";

    public static String formatDate(String date, String inputFormat, String outputFormat){
        if (date != null && !date.isEmpty()) {
            try {
                SimpleDateFormat inFormat = new SimpleDateFormat(inputFormat);
                Date newDate = inFormat.parse(date);

                SimpleDateFormat outformat = new SimpleDateFormat(outputFormat);
                return outformat.format(newDate);
            } catch (ParseException e) {
                return null;
            }
        } else {
          return null;
        }
    }
}

File: app/src/main/java/com/icerrate/popularmovies/utils/FormatUtils.kt
package com.icerrate.popularmovies.utils

import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

/**
 * @author Ivan Cerrate.
 */
object FormatUtils {

    const val FORMAT_YYYY_MM_DD = "yyyy-MM-dd"
    const val FORMAT_MMM_DD_YYYY = "MMM', 'dd' 'yyyy"

    fun formatDate(date: String?, inputFormat: String, outputFormat: String): String {
        return if (!date.isNullOrEmpty()) {
            try {
                val inFormat = SimpleDateFormat(inputFormat, Locale.getDefault())
                val newDate = inFormat.parse(date)
                val outFormat = SimpleDateFormat(outputFormat, Locale.getDefault())
                outFormat.format(newDate!!)
            } catch (e: ParseException) {
                "-"
            }
        } else {
            "-"
        }
    }
}